packages/react-databrowser/src/components/databrowser/components/data-display-container/data-delete.tsx

```diff
@@ -0,0 +1,50 @@
+import { Button } from "@/components/ui/button";
+import { useToast } from "@/components/ui/use-toast";
+import { MinusCircledIcon } from "@radix-ui/react-icons";
+import { useDeleteKey } from "../../hooks/useDeleteKey";
+import { DeleteAlertDialog } from "./delete-alert-dialog";
+import { RedisDataTypeUnion } from "@/types";
+
+type Props = {
+  selectedDataKey: string;
+  onDataKeyChange: (dataKey?: [string, RedisDataTypeUnion]) => void;
+};
+
+export const DataDelete = ({ onDataKeyChange, selectedDataKey }: Props) => {
+  const { toast } = useToast();
+  const deleteKey = useDeleteKey();
+
+  const handleDeleteKey = async () => {
+    try {
+      const result = await deleteKey.mutateAsync(selectedDataKey);
+      if (result) {
+        onDataKeyChange(undefined);
```

Comment:
did we forget this as undefined?

Reply:
You need to reset a chosen key to refresh it from the sidebar. When you delete a key, the selected item disappears from the left side, which then resets the right side. You can even try it out yourself.